python/ray/__init__.py

```diff
@@ -257,5 +260,18 @@ def __getattr__(self, attr):
 ]
 
 
+if sys.version_info < (3, 7):
+    # TODO(Clark): Remove this one we drop Python 3.6 support.
+    from ray import autoscaler  # noqa: F401
+    from ray import data  # noqa: F401
+    from ray import workflow  # noqa: F401
```

Comment:
We could implement something similar for Python 3.6 via a "replace module with class hack":
```python
class module(types.ModuleType):
    __all__ = __all__

    def __init__(self, orig_module: types.ModuleType):
        self._orig_module = orig_module

    def __getattr__(self, name: str):
        try:
            return getattr(name, self._orig_module)
        except AttributeError as e:
            if name in _subpackages:
                return importlib.import_module("." + name, self._orig_module.__name__)
            raise e from None


sys.modules[__name__] = module(sys.modules[__name__])
```

But IMO this is more hacky than its worth since we're going to be dropping Python 3.6 support in the near-term.